spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -491,10 +491,10 @@ public void write(InternalRow row) throws IOException {
         AppenderFactory<InternalRow> appenderFactory,
         WriterFactory.OutputFileFactory fileFactory,
         FileIO fileIo,
-        long targetFileSize) {
+        long targetFileSize,
+        Schema schema) {
```

Comment:
Can we call this `writeSchema` or some name that indicates it is the schema of the incoming data, not necessarily the table schema?